test/runtests.py

```diff
@@ -7,114 +7,208 @@
 import sys
 import os
 import subprocess as SP
+import argparse
+import xml.etree.ElementTree as ET
 
-test_all = True
-test_root = os.path.dirname(os.path.realpath(__file__))
+# handle command line args
+parser = argparse.ArgumentParser(
+    description='ChakraCore *nix Test Script',
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    epilog='''\
+Samples:
+
+test all folders:
+    {0}
+
+test only Array:
+    {0} Array
+
+test a single file:
+    {0} Basics/hello.js
+'''.format(sys.argv[0]))
+
+parser.add_argument('folders', metavar='folder', nargs='*',
+                    help='folder subset to run tests')
+parser.add_argument('-b', '--binary', metavar='bin', help='ch full path')
+parser.add_argument('-d', '--debug', action='store_true',
+                    help='use debug build');
+parser.add_argument('-t', '--test', action='store_true', help='use test build')
+parser.add_argument('--x86', action='store_true', help='use x86 build')
+parser.add_argument('--x64', action='store_true', help='use x64 build')
+args = parser.parse_args()
 
-# ugly trick
-ch_path = os.path.join(os.path.dirname(test_root), "BuildLinux/ch")
 
-if not os.path.isfile(ch_path):
-    print "BuildLinux/ch not found. Did you run ./build.sh already?"
+test_root = os.path.dirname(os.path.realpath(__file__))
+repo_root = os.path.dirname(test_root)
+
+# arch: x86, x64
+arch = 'x86' if args.x86 else ('x64' if args.x64 else None)
+if arch == None:
+    arch = os.environ.get('_BuildArch', 'x86')
+
+# flavor: debug, test, release
+type_flavor = {'chk':'debug', 'test':'test', 'fre':'release'}
+flavor = 'debug' if args.debug else ('test' if args.test else None)
+if flavor == None:
+    flavor = type_flavor[os.environ.get('_BuildType', 'fre')]
```

Comment:
> If this is hardcoded, why have the map lookup?

It is not hardcoded. It reads env variable `_BuildType`, and default to `fre` if no env variable.